disabled-tests-inspector/src/main/java/io/quarkus/ts/DisabledTestAnalyserService.java

```diff
@@ -0,0 +1,217 @@
+package io.quarkus.ts;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.enterprise.context.ApplicationScoped;
+
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GHTree;
+import org.kohsuke.github.GHTreeEntry;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHIssueState;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@ApplicationScoped
+public class DisabledTestAnalyserService {
+
+    private static final Pattern CLASS_DECLARATION_PATTERN = Pattern.compile(
+            "public\\s+(?:\\w+\\s+)*class\\s+(\\w+)"
+    );
+
+    private static final Pattern TEST_METHOD_PATTERN = Pattern.compile(
+            "public\\s+void\\s+(\\w+)\\s*\\("
+    );
+
+    private static final Pattern DISABLED_ANNOTATION_PATTERN = Pattern.compile(
+            "@(Disabled\\w*)\\s*(?:\\((.*)\\))?"
+    );
+
+    private static final Pattern REASON_PATTERN = Pattern.compile(
+            "reason\\s*=\\s*\"([^\"]+)\"", Pattern.CASE_INSENSITIVE
+    );
+
+    private static final Pattern EXPLICIT_REASON_PATTERN = Pattern.compile(
+            "\"(.*)\""
+    );
+
+    private static final Pattern ISSUE_PATTERN = Pattern.compile(
+    "(https://(?:github\\.com/.+?/issues/\\d+|issues\\.redhat\\.com/browse/\\w+-\\d+))"
+    );
+
+
+    public void analyzeRepository(String repoOwner, String repoName, List<String> branches, String baseOutputFileName) throws IOException {
+        GitHub github = GitHub.connect();
+        GHRepository repo = github.getRepository(repoOwner + "/" + repoName);
+
+        for (String branch : branches) {
+            GHTree tree = repo.getTreeRecursive(branch, 1);
+            List<DisabledTest> disabledTests = new ArrayList<>();
+            Map<String, DisabledTestsModuleStats> moduleStats = new HashMap<>();
+
+            for (GHTreeEntry entry : tree.getTree()) {
+                String filePath = entry.getPath();
+                if (entry.getPath().contains("/test/") && filePath.endsWith(".java")) {
+                    GHContent fileContent = repo.getFileContent(filePath, branch);
```

Comment:
this makes remote request per every java file in a test root.  opening and closing connection for every file is time consuming (it's like difference between loading stuff in chunks, so that you avoid open/close). It kinda depends on how well is git implemented, because spare checkout allows you to specify pattern, so then you would just specify patterns for test java files and fetch it together. but if they don't download it in chunk, it wouldn't improve anything.